modules/deployment-model-impl/src/main/java/org/mule/runtime/module/deployment/impl/internal/policy/DefaultPolicyInstance.java

```diff
@@ -0,0 +1,157 @@
+                               PolicyParametrization parametrization, ServiceRepository serviceRepository,
+                               ClassLoaderRepository classLoaderRepository) {
+    this.application = application;
+    this.template = template;
+    this.parametrization = parametrization;
+    this.serviceRepository = serviceRepository;
+    this.classLoaderRepository = classLoaderRepository;
+  }
+
+  private void initPolicyContext() {
+    if (policyContext == null) {
+      ArtifactContextBuilder artifactBuilder =
+          newBuilder().setArtifactType(APP)
+              .setArtifactName(parametrization.getId())
+              .setConfigurationFiles(getResourcePaths(template.getDescriptor().getConfigResourceFiles()))
+              .setExecutionClassloader(template.getArtifactClassLoader().getClassLoader())
+              .setServiceRepository(serviceRepository)
+              .setClassLoaderRepository(classLoaderRepository);
+
+      artifactBuilder.withServiceConfigurator(customizationService -> customizationService
+          .overrideDefaultServiceImpl(MuleProperties.OBJECT_POLICY_MANAGER_STATE_HANDLER,
+                                      application.getMuleContext().getRegistry()
+                                          .lookupObject(MuleProperties.OBJECT_POLICY_MANAGER_STATE_HANDLER)));
+
+      try {
+        policyContext = artifactBuilder.build();
+        policyContext.getMuleContext().start();
+      } catch (MuleException e) {
+        throw new IllegalStateException("Cannot create artifact context for the policy instance", e);
+      }
+    }
+  }
+
+  private String[] getResourcePaths(File[] configResourceFiles) {
+    List<String> paths = new ArrayList<>();
+    for (File configResourceFile : configResourceFiles) {
+      paths.add(configResourceFile.getAbsolutePath());
+    }
+
+    return paths.toArray(new String[0]);
+  }
+
+  @Override
+  public List<Policy> findSourceParameterizedPolicies(
+                                                      PolicyPointcutParameters policyPointcutParameters) {
+
+    initPolicyInstance();
+
+    return policyInstance.getSourcePolicyChain()
+        .map(operationChain -> asList(new Policy(operationChain, parametrization.getId())))
+        .orElse(emptyList());
+  }
+
+  private void initPolicyInstance() {
+
+    if (policyInstance == null) {
+      synchronized (this) {
+        initPolicyContext();
+
+        try {
+          policyInstance = policyContext.getMuleContext().getRegistry().lookupObject(
+                                                                                     org.mule.runtime.core.policy.DefaultPolicyInstance.class);
+        } catch (RegistrationException e) {
+          throw new IllegalStateException(String.format("More than one %s found on context", PolicyInstance.class), e);
+        }
+
+        try {
+          policyInstance.initialise();
+          policyInstance.start();
```

Comment:
how do you know you should start it if this class is not lifecycle aware? If this is coming from the registry, then mule should be in charge of applying lifecycle